### PR TITLE
Invoke-DbaSqlQuery, enhance messaging

### PR DESCRIPTION
**Pull request by niphlod**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [x] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Tinker easily with queries where messages (PRINT and RAISERROR) are as important as the result
With stream redirection it has almost been possible to do this but it's always kinda be a PITA. This has been discussed in slack for a while, sorry for the delay

### Approach
Basically stripped down the code we need from invoke-sqlcmd2 and adapted to use dbatools' facilities, coupled with async execution and a good tip from @FriedrichWeinmann. It's safe to assume we can now move invoke-sqlcmd2 to the optional folder, as this, coupled with connect-dbainstance to establish "funny" connections, has all the needed features.

### Commands to test
Added basic tests to make sure output is not buffered (as it is in invoke-sqlcmd and sqlcmd -ge 2012). Think reading the output of ola's maintenance procedures while it is outputted without waiting for the procedure to end.

### CODE REVIEW

**Comment by potatoqualitee** [comment]
oh la la! ty will look when i have time and a brain.
